### PR TITLE
R4R: QueryEmojiRank

### PR DESCRIPTION
**Pull request by yaruwangway**
add feature `QueryEmojiRank`
try command `pooltoy q faucet  emoji-rank [num]` to show top num emoji holders

### CODE REVIEW

**Comment by yaruwangway** [comment]
commit e4c652feb6aa7a258ff4e57aaa2b593e8e8ff5ef fixed the issue of parsing emoji in command  `pooltoy tx faucet mintfor`.  Please refer to the linked issues for more details.

**Comment by okwme** [comment]
is this R4R or still in draft mode? let's use "DRAFT" or "R4R" in the title to designate?

**Comment by yaruwangway** [comment]
OK! It is ready!

File: app/app.go [hunk 1]
```diff
@@ -183,8 +183,8 @@ type App struct {
 	BankKeeper       bankkeeper.Keeper
 	CapabilityKeeper *capabilitykeeper.Keeper
 	StakingKeeper    stakingkeeper.Keeper
-	SlashingKeeper   slashingkeeper.Keeper
-	MintKeeper       mintkeeper.Keeper
+	SlashingKeeper   slashingkeeper.Keeper  // todo check if needed
```
Comment (okwme) [inline app/app.go:186]:
```suggestion
	SlashingKeeper   slashingkeeper.Keeper
```

File: app/app.go [hunk 2]
```diff
@@ -183,8 +183,8 @@ type App struct {
...
+	MintKeeper       mintkeeper.Keeper // todo check if needed
```
Comment (okwme) [inline app/app.go:187]:
```suggestion
	MintKeeper       mintkeeper.Keeper
```

**Review by okwme — APPROVED** [review]

File: app/ante.go [hunk 3]
```diff
@@ -9,7 +9,7 @@ import (
 	"github.com/cosmos/cosmos-sdk/x/auth/types"
 	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
 )
-
+// todo why this decorator has the same keeper as the keeper?
```
Comment (okwme) [inline app/ante.go:12]:
```suggestion
```

**Review by okwme — APPROVED** [review]

**Pull request merged**